compiler/rustc_target/src/spec/windows_gnullvm_base.rs

```diff
@@ -0,0 +1,50 @@
+use crate::spec::{cvs, LinkArgs, LinkerFlavor, TargetOptions};
+
+pub fn opts() -> TargetOptions {
+    let pre_link_args = LinkArgs::from([(
+        LinkerFlavor::Gcc,
+        vec![
+            // We cannot use `-nodefaultlibs` because compiler-rt has to be
+            // passed as a path.
+            "-nolibc".into(),
+            "--unwindlib=none".into(),
```

Comment:
So `-nolibc --unwindlib=none` is basically how `-nodefaultlibs` looks for this target.
Then maybe it should go into `fn no_default_libraries` in `linker.rs`.

>because compiler-rt has to be passed as a path

Also, could you tell about this in some more detail? Why is this target special in this regard?

Reply:
> > because compiler-rt has to be passed as a path
> 
> Also, could you tell about this in some more detail? Why is this target special in this regard?

With `libgcc` this is rather obvious: path like `/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/` is added via `-L` so one can use `-lgcc` and the linker will have no problems looking for it (Clang also searches for GCC libs).

`compiler-rt` however is special here (not in good meaning), it's located in Clang lib dir similarly to `libgcc` so on Linux it's `/usr/lib/clang/13.0.1/lib/linux` but this path is not added via `-L`.  Instead Clang will pass absolute path of the library to the linker (unless using some `-no*` options).
Unfortunately LLVM devs are not in favour of fixing it to behave like `libgcc`: https://reviews.llvm.org/D51440

Alternatively we could call `clang --print-resource-dir` and add it to linker path in every linking process or ship our own compiler-rt build which might be hard to keep compatible with different compiler versions.

Reply:
Could you add this information to the comment?

Reply:
> So -nolibc --unwindlib=none is basically how -nodefaultlibs looks for this target.
> Then maybe it should go into fn no_default_libraries in linker.rs.

IMO it's not worth doing it right now (complicating the other code) for a tier 3 target.

> Could you add this information to the comment?

I think it should be fine now.